### PR TITLE
feat: Task & Manager

### PR DESCRIPTION
**Pull request by gavin1ee**
Task CURD, StartTask, StopTask, SchedulerServerManager regist and assign Task to worker.  SchedulerWorkerManager regist and run HttpJob

### CODE REVIEW

File: src/ApiGateways/Caller/Masa.Scheduler.ApiGateways.Caller/Services/SchedulerJobService.cs [hunk 1]
```diff
@@ -15,7 +15,7 @@ public SchedulerJobService(ICallerProvider callerProvider) : base(callerProvider
     public async Task<SchedulerJobListResponse> GetListAsync(SchedulerJobListRequest request)
     {
         var result = await GetAsync<SchedulerJobListRequest,SchedulerJobListResponse>(string.Empty, request);
-        return result ?? new();
+        return result ?? new(0, 0, new());
```
Comment (MayueCif) [inline src/ApiGateways/Caller/Masa.Scheduler.ApiGateways.Caller/Services/SchedulerJobService.cs:18]: new(0, 0, new()) ==? new(new())

File: src/ApiGateways/Caller/Masa.Scheduler.ApiGateways.Caller/Services/SchedulerJobService.cs [hunk 2]
```diff
@@ -15,7 +15,7 @@ public SchedulerJobService(ICallerProvider callerProvider) : base(callerProvider
     public async Task<SchedulerJobListResponse> GetListAsync(SchedulerJobListRequest request)
     {
         var result = await GetAsync<SchedulerJobListRequest,SchedulerJobListResponse>(string.Empty, request);
```
Comment (MayueCif) [inline src/ApiGateways/Caller/Masa.Scheduler.ApiGateways.Caller/Services/SchedulerJobService.cs:17]: string.Empty ?

File: src/Contracts/MASA.Scheduler.Contracts.Server/Dtos/SchedulerTaskDto.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+// Licensed under the Apache License. See LICENSE.txt in the project root for license information.
+
+namespace Masa.Scheduler.Contracts.Server.Dtos;
+
+public class SchedulerTaskDto
+{
+    public Guid Id { get; set; }
+
+    public int RunCount { get; set; }
+
```
Comment (MayueCif) [inline src/Contracts/MASA.Scheduler.Contracts.Server/Dtos/SchedulerTaskDto.cs:11]: comment minutes or seconds?

File: src/Contracts/MASA.Scheduler.Contracts.Server/Infrastructure/IntegrationEvents/BaseIntegrationEvent.cs [hunk 4]
```diff
@@ -0,0 +1,9 @@
+﻿// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the Apache License. See LICENSE.txt in the project root for license information.
+
+namespace Masa.Scheduler.Contracts.Server.Infrastructure.IntegrationEvents;
+
+public abstract record BaseIntegrationEvent : IntegrationEvent
+{
+    public Guid ProgramId { get; set; }
```
Comment (MayueCif) [inline src/Contracts/MASA.Scheduler.Contracts.Server/Infrastructure/IntegrationEvents/BaseIntegrationEvent.cs:8]: ProgramId?

File: src/Contracts/MASA.Scheduler.Contracts.Server/Infrastructure/IntegrationEvents/OnlineIntegrationEvent.cs [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+{
+    public string HttpHost { get; set; } = string.Empty;
+
+    public string HttpsHost { get; set; } = string.Empty;
+
+    public int HttpPort { get; set; }
+
+    public int HttpsPort { get; set; }
+
+    public bool IsResponse { get; set; }
```
Comment (MayueCif) [inline src/Contracts/MASA.Scheduler.Contracts.Server/Infrastructure/IntegrationEvents/OnlineIntegrationEvent.cs:16]: IsPong

File: src/Contracts/MASA.Scheduler.Contracts.Server/Model/WorkerModel.cs [hunk 6]
```diff
@@ -0,0 +1,14 @@
+﻿// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the Apache License. See LICENSE.txt in the project root for license information.
+
+namespace Masa.Scheduler.Contracts.Server.Model;
+
+public class WorkerModel : BaseServiceModel
+{
+    public int CurrentRunTaskCount { get; set; }
```
Comment (MayueCif) [inline src/Contracts/MASA.Scheduler.Contracts.Server/Model/WorkerModel.cs:8]: CurrentTaskRunCount?

File: src/Contracts/MASA.Scheduler.Contracts.Server/Requests/SchedulerTasks/NotifySchedulerTaskRunResultRequest.cs [hunk 7]
```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the Apache License. See LICENSE.txt in the project root for license information.
+
+namespace Masa.Scheduler.Contracts.Server.Requests.SchedulerTasks;
+
+public class NotifySchedulerTaskRunResultRequest : BaseRequest
+{
+    public Guid TaskId { get; set; }
+
+    public bool IsSuccess { get; set; }
```
Comment (MayueCif) [inline src/Contracts/MASA.Scheduler.Contracts.Server/Requests/SchedulerTasks/NotifySchedulerTaskRunResultRequest.cs:10]: enum replace IsSuccess and IsCancel

File: src/Contracts/MASA.Scheduler.Contracts.Server/Model/BaseServiceModel.cs [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+
+    public int NotResponseCount { get; set; }
+
+    public string HeartbeatApi { get; set; } = string.Empty;
+
+    public HttpClient CallerClient { get; set; } = default!;
+
+    public string GetServiceUrl(bool containsScheme = true)
+    {
+        if (HttpsPort != 0)
```
Comment (MayueCif) [inline src/Contracts/MASA.Scheduler.Contracts.Server/Model/BaseServiceModel.cs:30]: HttpsPort  has value but HttpsHost no value,HttpsHost  should main condition and handle 80、443

File: src/Services/Masa.Scheduler.Services.Server/Domain/EventHandler/NotifyTaskRunResultDomainEventHandler.cs [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    [EventHandler]
+    public async Task NotifyTaskRunResultAsync(NotifyTaskRunResultDomainEvent @event)
+    {
+        var task = await _dbContext.Tasks.Include(t => t.Job).FirstOrDefaultAsync(t => t.Id == @event.Request.TaskId);
+
+        if (task == null)
+        {
+            return;
```
Comment (MayueCif) [inline src/Services/Masa.Scheduler.Services.Server/Domain/EventHandler/NotifyTaskRunResultDomainEventHandler.cs:26]: need a prompt？

**Review by MayueCif — COMMENTED** [review]
modify by conversation

**Review by MayueCif — APPROVED** [review]

**Comment by MayueCif** [comment]
@doddgu LGTM

File: src/Services/Masa.Scheduler.Services.Server/Application/Tasks/Commands/StartSchedulerTaskCommand.cs [hunk 10]
```diff
@@ -0,0 +1,6 @@
+﻿// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the Apache License. See LICENSE.txt in the project root for license information.
+
+namespace Masa.Scheduler.Services.Server.Application.Tasks.Commands;
+
+public record StartSchedulerTaskCommand (StartSchedulerTaskRequest Request) : Command;
```
Comment (doddgu) [inline src/Services/Masa.Scheduler.Services.Server/Application/Tasks/Commands/StartSchedulerTaskCommand.cs:6]: What's the different between `SchedulerTaskStartCommand` and `StartSchedulerTaskCommand`?

**Review by doddgu — CHANGES_REQUESTED** [review]

File: src/Services/Masa.Scheduler.Services.Server/Application/Tasks/Commands/StartSchedulerTaskCommand.cs (hunk 10 above)
Comment (gavin1ee) [reply src/Services/Masa.Scheduler.Services.Server/Application/Tasks/Commands/StartSchedulerTaskCommand.cs:6]: SchedulerTaskStartCommand  是当Worker通知Server某个Task 启动时. Server的处理command.  
StartSchedulerTaskCommand  是当用于在页面上点击运行job/运行任务时. Server的处理command. 实际上Task还未真正开始运行. 只是把Task 丢给了Worker. 由Worker 运行.
目前为了防止混淆.  已经将SchedulerTaskStartCommand  改成了 NotifyTaskStartCommand

**Review by doddgu — APPROVED** [review]
LGTM

**Pull request merged**